### PR TITLE
[SC-102] Add Arbitrum and Optimism forwarders and executors

### PR DESCRIPTION
**Pull request by barrutko**
Proposed directory structure (_updated_):
- 
- src
  - dependencies
    - arbitrum
      - AddressAliasHelper.sol [_governance-crosschain-bridges/contracts/dependencies/arbitrum_]
      - interfaces
        - IBridge.sol [_governance-crosschain-bridges/contracts/dependencies/arbitrum/interfaces_]
        - IDelayedMessageProvider.sol [_governance-crosschain-bridges/contracts/dependencies/arbitrum/interfaces_]
        - IGasRefunder.sol [_governance-crosschain-bridges/contracts/dependencies/arbitrum/libraries_]
        - IInbox.sol [_governance-crosschain-bridges/contracts/dependencies/arbitrum/interfaces_]
        - IOwnable.sol [_governance-crosschain-bridges/contracts/dependencies/arbitrum/interfaces_]
        - ISequencerInbox.sol [_governance-crosschain-bridges/contracts/dependencies/arbitrum/interfaces_]
    - optimism
      - interfaces
        - ICrossDomainMessenger.sol [_governance-crosschain-bridges/contracts/dependencies/optimism/interfaces_]
        - IL2CrossDomainMessenger.sol [_governance-crosschain-bridges/contracts/dependencies/optimism/interfaces_]
  - executors
    - BridgeExecutorBase.sol [_governance-crosschain-bridges/contracts/bridges_]
    - L2BridgeExecutor.sol [_governance-crosschain-bridges/contracts/bridges_]
    - ArbitrumBridgeExecutor.sol [_governance-crosschain-bridges/contracts/bridges_]
    - OptimismBridgeExecutor.sol [_governance-crosschain-bridges/contracts/bridges_]
  - forwarders
    - CrosschainForwarderArbitrum.sol [_aave-helpers/src/crosschainforwarders_]
    - CrosschainForwarderOptimism.sol [_aave-helpers/src/crosschainforwarders_]
  - interfaces
    - IExecutorBase.sol [_governance-crosschain-bridges/contracts//interfaces_]
    - IL2BridgeExecutor.sol [_governance-crosschain-bridges/contracts//interfaces_]
    - IProposalGenericExecutor.sol [_aave-helpers/src/interfaces_]
- tests
  - mocks
    - PayloadWithEmit.sol [_aave-helpers/tests/mocks_]
  - ArbitrumCrosschainForwarderTest.t.sol [_aave-helpers/tests/crosschainforwarders_]
  - OptimismCrosschainForwarderTest.t.sol [_aave-helpers/tests/crosschainforwarders_]

### CODE REVIEW

File: .gitmodules [hunk 1]
```diff
@@ -1,3 +1,6 @@
 [submodule "lib/forge-std"]
 	path = lib/forge-std
 	url = https://github.com/foundry-rs/forge-std
+[submodule "lib/aave-helpers"]
+	path = lib/aave-helpers
+	url = https://github.com/marsfoundation/aave-helpers
```
Comment (hexonaut) [inline .gitmodules:6]: So I see this is only being used for the tests. I think it's okay for this initial PR, but as a next step we probably want to remove this an make the test code generic (no `GovHelpers`, proposals, etc) using the`xchain-helpers` repo for the e2e testing.

**Review by hexonaut — CHANGES_REQUESTED** [review]
Before continuing I think this should be restructured slightly. Also there is missing newlines at EOF. You should configure Visual Studio Code or whatever your IDE is to add these automatically.

Proposed Adjustments:

 * Move `BridgeExecutorBase` and `L2BridgeExecutor` into `executors` dir.
 * Create two directories in `dependencies` called `optimism` and `arbitrum` to put all the chain-specific dependencies (including interfaces)
 * Move `interfaces/(arbitrum|optimism)` into `dependencies/(arbitrum|optimism)/interfaces`
 * Move `interfaces/executors` up one directory to `interfaces`
 * Move `AddressAliasHelper` into `dependencies/arbitrum` as it's specific to Arbitrum.

File: src/dependencies/arbitrum/interfaces/IGasRefunder.sol [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+interface IGasRefunder {
+  function onGasSpent(
+    address payable spender,
+    uint256 gasUsed,
+    uint256 calldataSize
+  ) external returns (bool success);
+}
+
+abstract contract GasRefundEnabled {
```
Comment (hexonaut) [inline src/dependencies/arbitrum/interfaces/IGasRefunder.sol]: This isn't an interface and also isn't even used anywhere. Can just remove it I think.

File: src/interfaces/IProposalGenericExecutor.sol [hunk 3]
```diff
@@ -0,0 +1,9 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+/**
+ * @dev Generic proposal interface allowing execution via MockExecutor
+ */
+interface IProposalGenericExecutor {
+  function execute() external;
+}
```
Comment (hexonaut) [inline src/interfaces/IProposalGenericExecutor.sol]: This is just used for testing (and will probably be removed in the next PR). I would inline it in the `PayloadWithEmit` for now so we don't forget to remove later.

**Review by hexonaut — CHANGES_REQUESTED** [review]
Minor changes, all code matches source repos. After they are addressed this looks good for merging. I don't care too much about the state of the tests rn as they should be rewritten to be generic and use `xchain-helpers` instead of manually doing the message relaying in the e2e tests BGD built.

File: src/dependencies/arbitrum/interfaces/IGasRefunder.sol (hunk 2 above)
Comment (barrutko) [reply src/dependencies/arbitrum/interfaces/IGasRefunder.sol]: Done ✅

File: src/interfaces/IProposalGenericExecutor.sol (hunk 3 above)
Comment (barrutko) [reply src/interfaces/IProposalGenericExecutor.sol]: Done ✅

**Review by hexonaut — APPROVED** [review]
LGTM

**Review by lucas-manuel — APPROVED** [review]
LGTM, lets merge as is, going to make another ticket to investigate how to structure GovHelpers cause it looks like its needed in both spells and here

**Pull request merged**